script.js

```diff
@@ -0,0 +1,44 @@
+let saladMenuFileData = [], prices = [];
+let bill = 0, divContent="<table>";
+
+fetch("/salads.json")
+  .then((response) => response.json())
+  .then((saladMenuFileData) => createMenu(saladMenuFileData));
+
+  function setPrice(counter,saladMenuFileData){
+    if((counter + 1) % 3 == 0) return 15;
+    else if(saladMenuFileData[counter][0] == 'E' ||
+            saladMenuFileData[counter][0] == 'R' ||
+            saladMenuFileData[counter][0] == 'G' ) return 10;
+    else return 20;
+  }
+
+function createMenu(saladMenuFileData){
+  let divMenu = document.getElementById("menu");
+    for(let indexCounter=0; indexCounter < saladMenuFileData.length; indexCounter++){
+      prices[indexCounter] = setPrice(indexCounter, saladMenuFileData);
+       divContent += "<tr  >";
+       divContent += "<td class='classContentRow' id=salad" + indexCounter + ">" + (indexCounter + 1);
+       divContent += ". " + saladMenuFileData[indexCounter] +"</td>";
+       divContent += "<td> " + prices[indexCounter] + "zł</td>";
+       divContent += "<td><input type='checkbox' id='checkbox" + indexCounter + "' value='" + indexCounter + "'onclick='checkMenuBox(id,value)'></td>";
+       divContent += "</tr>";
+    }
+  divContent+="</table>";
+  divMenu.innerHTML = divContent;
+}
+
+function checkMenuBox(boxId,boxValue){
+  boxValue = parseInt(boxValue);
+  if(document.getElementById(boxId).checked == true)
+  {
+    bill += prices[boxValue];
+    document.getElementById("salad"+boxValue).style = "font-weight:bold;";

```

Comment:
Jest bardzo przyjazna w konkatenowaniu stringów składania którą dostarcza es6. Template literals. W backquoatach możemy konkatenować stringi ze zemiennymi za pomocą ${<nazwa zmiennej>}. Przykład:

const flavor = "becon";
`My favorite ice cream flavor is ${}. Who would have thought!`

Reply:
A style trzymamy w osobnych plikach css. Inline style tak jak tutaj, są potrzebne tylko w przypadku dynamicznego stylowania (czyli podczas używania aplikacji). To ze względu na to, że inline styles ciężko nadpisać. Tutaj dokumentacja:
https://www.w3.org/TR/selectors-3/#specificity

I artykuł jakby coś było nie jasne:
https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity